repos/system_upgrade/el8toel9/actors/multipathconfread/tests/test_multipath_conf_read_8to9.py

```diff
@@ -0,0 +1,182 @@
+
+
+def build_config(val):
+    return MultipathConfig8to9(
+        pathname=val[0],
+        config_dir=val[1],
+        enable_foreign_exists=val[2],
+        invalid_regexes_exist=val[3],
+        allow_usb_exists=val[4],
+    )
+
+
+def assert_config(config, expected):
+    assert config.pathname == expected.pathname
+    assert config.config_dir == expected.config_dir
+    assert config.enable_foreign_exists == expected.enable_foreign_exists
+    assert config.invalid_regexes_exist == expected.invalid_regexes_exist
+    assert config.allow_usb_exists == expected.allow_usb_exists
+
+
+default_rhel8_conf = build_config(
+    [os.path.join(TEST_DIR, 'default_rhel8.conf'), None, True, False, False, ])
+
+all_the_things_conf = build_config(
+    [os.path.join(TEST_DIR, 'all_the_things.conf'), None, False, True, False, ])
+
+converted_the_things_conf = build_config(
+    [os.path.join(TEST_DIR, 'converted_the_things.conf'), None, True, False, True, ])
+
+complicated_conf = build_config(
+    [os.path.join(TEST_DIR, 'complicated.conf'), "/etc/multipath/conf.d", True, True, False, ])
+
+no_foreign_conf = build_config(
+    [os.path.join(TEST_DIR, 'no_foreign.conf'), None, False, True, True, ])
+
+allow_usb_conf = build_config(
+    [os.path.join(TEST_DIR, 'allow_usb.conf'), None, False, False, True, ])
+
+empty_conf = build_config(
+    [os.path.join(TEST_DIR, 'empty.conf'), None, False, False, False, ])
+
+missing_dir_conf = build_config(
+    [os.path.join(TEST_DIR, 'missing_dir.conf'), os.path.join(TEST_DIR, 'missing'), False, True, False, ])
+
+empty_dir_conf = build_config(
+    [os.path.join(TEST_DIR, 'empty_dir.conf'), os.path.join(TEST_DIR, 'conf3.d'), False, False, False, ])
+
+not_set_dir_conf = build_config(
+    [os.path.join(TEST_DIR, 'not_set_dir.conf'), os.path.join(TEST_DIR, "conf1.d"), False, True, False, ])
+
+empty1_conf = build_config(
+    [os.path.join(TEST_DIR, 'conf1.d/empty.conf'), None, False, False, False, ])
+
+nothing_important_conf = build_config(
+    [os.path.join(TEST_DIR, 'conf1.d/nothing_important.conf'),
+     os.path.join(TEST_DIR, 'this_gets_ignored'), False, False, False, ])
+
+set_in_dir_conf = build_config(
+    [os.path.join(TEST_DIR, 'set_in_dir.conf'), os.path.join(TEST_DIR, "conf2.d"), False, False, False, ])
+
+all_true_conf = build_config(
+    [os.path.join(TEST_DIR, 'conf2.d/all_true.conf'), None, True, True, True, ])
+
+no_defaults_conf = build_config(
+    [os.path.join(TEST_DIR, 'no_defaults.conf'), None, False, True, False, ])
+
+two_defaults_conf = build_config(
+    [os.path.join(TEST_DIR, 'two_defaults.conf'), None, True, False, False, ])
+
+
+def mock_parse_config(path):
+    """Convert config_dir into full pathname"""
+    conf = multipathconfread._parse_config_orig(path)
+    if not conf:
+        return None
+    if conf.config_dir:
+        conf.config_dir = os.path.join(TEST_DIR, conf.config_dir)
+    return conf
+
+
+def test_parse_default_rhel8():
```

Comment:
Tests L87-L138 seem very much alike. How about introducing a mapping like
```
test_map = {'default_rhel8.conf': default_rhel8_conf, 
            'converted_the_things.conf': converted_the_things_conf, 
            'compicated.conf': complicated_conf, ...}
```
And the have a single test
```
def_test_parse_configs():
    for config_name, expected_data in test_mapping.items():
        config = multipathconfread._parse_config(os.path.join(TEST_DIR, config_name))
        assert config
        assert_config(config, expected_data)
```
WDYT?

Reply:
That sounds fine.